src/mono/browser/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -147,10 +147,23 @@ protected override async Task<bool> AcceptEvent(SessionId sessionId, JObject par
                     }
                 case "Debugger.scriptParsed":
                     {
-                        if (args["url"]?.ToString()?.Contains("/_framework/") == true) //is from dotnet runtime framework
+                        if (!Contexts.TryGetCurrentExecutionContextValue(sessionId, out ExecutionContext context))
+                            return false;
+                        try
                         {
-                            if (Contexts.TryGetCurrentExecutionContextValue(sessionId, out ExecutionContext context))
+                            var callStack = args["stackTrace"]?["callFrames"]?.Value<JArray>();
+                            var topFrameFunctionName = callStack?.Count > 0 ? callStack?[0]?["functionName"]?.Value<string>() : null;
+                            //skip mono_wasm_fire_debugger_agent_message_with_data_to_pause or mono_wasm_runtime_ready (both of them have debugger; statement)
+                            if (args["url"]?.ToString()?.Equals("") == true && topFrameFunctionName?.StartsWith("mono_wasm_", StringComparison.Ordinal) == true)
```

Comment:
nit: 
```suggestion
                            if (args["url"]?.ToString()?.Equals("") == true && topFrameFunctionName?.StartsWith("mono_wasm_", StringComparison.OrdinalIgnoreCase) == true)
```

Reply:
Also, `if (args["url"]?.ToString()?.Equals("") == true ` this can be checked before getting the `stackTrace` to return early.

maybe: `if (string.IsNullOrEmpty(args["url"]?.ToString())`